core/dbt/task/compare.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import print_function
+
+from dbt.adapters.factory import get_adapter
+from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.runner import RunManager
+from dbt.node_types import NodeType
+from dbt.node_runners import ModelRunner
+
+import dbt.ui.printer
+from dbt.task.compile import CompileTask
+
+class CompareTask(CompileTask):
+    def _get_manifest(self):
+        compiler = dbt.compilation.Compiler(self.config)
+        compiler.initialize()
+
+        all_projects = compiler.get_all_projects()
+
+        manifest = dbt.loader.GraphLoader.load_all(self.config, all_projects)
+        return manifest
+
+    def run(self):
+
+        logger.info('dbt compare: Comparing local codebase to catalog in data warehouse.')
+
+        manifest = self._get_manifest()
+
+        # Look up all of the relations dbt knows about
+        used_schemas = manifest.get_used_schemas()
+        used_relations = []
+        for node in manifest.nodes.items():
```

Comment:
I updated to

Reply:
Hmmm, I brought up ephemeral models because it's pretty typical to reconfigure eg. a view model to become an ephemeral model. When that happens, there's an orphaned `view` which will never be updated by dbt. I think that `dbt compare` could be really useful in determining when that happens!

So, while the ephemeral model is indeed _in_ the manifest, it won't actually be rendered out into the db. I think that makes it fall into your first point:
> it looks for things in the database that are not present in the manifest

We might as well throw away any ephemeral models (as we do with disabled models) when computing the difference between the catalog and the manifest i think.

This is a huge part of why i was so hesitant to make dbt actually drop the orphaned relations in the database. You can imagine code like:

```
{% if target.name == 'prod' %}

-- really complicated and involved incremental logic here
-- for a massive table that you don't want to build in dev

{% else %}

{{ config(materialized='ephemeral') }}

select * from prod.big_table

{% endif %}
```

If you're not careful with your targets, then dbt could errantly tell you to drop this model!

Reply:
@drewbanin got it! This makes sense. Will implement.

Reply:
Okay, I believe this is done now. On to the tests...